apps/dotcms-ui/src/app/view/components/_common/dot-page-selector/dot-page-selector.component.ts

```diff
@@ -151,61 +157,111 @@ export class DotPageSelectorComponent implements ControlValueAccessor {
+                  tap(() => {
+                      this.searchType = 'site';
+                  })
+              )
+            : this.getSecondStepData(param);
     }
 
-    private resetResults(): void {
-        this.results = {
-            data: [],
-            query: '',
-            type: ''
-        };
+    private getSecondStepData(param: string): Observable<DotPageSelectorItem[]> {
+        return this.folderSearch
+            ? this.dotPageSelectorService.getFolders(param).pipe(
+                  tap(() => {
+                      this.searchType = 'folder';
+                  })
+              )
+            : this.dotPageSelectorService.getPages(param).pipe(
+                  tap(() => {
+                      this.searchType = 'page';
+                  })
+              );
     }
 
-    private getErrorMessage(): string {
-        return this.results.data.length === 0
-            ? this.results.type === 'site'
-                ? this.dotMessageService.get('page.selector.no.sites.results')
-                : this.dotMessageService.get('page.selector.no.page.results')
-            : null;
+    private isTwoStepSearch(param: string): boolean {
+        return (
+            param.startsWith('//') &&
+            param.length > 2 &&
+            (this.isHostAndPath(param) || param.endsWith('/'))
+        );
     }
 
-    private isOnlyFullHost(host: string): boolean {
-        return /\/\/[-a-zA-Z0-9@:%._\+~#=]{2,256}\.[a-z]{2,6}\/*$/.test(host);
+    private isHostAndPath(param: string): boolean {
+        const url: DotSimpleURL | { [key: string]: string } = this.parseUrl(param);
+        return url && !!(url.host && url.pathname.length > 0);
     }
 
-    private autoSelectUniqueResult(): void {
-        this.onSelect(this.results.data[0]);
+    private parseUrl(query: string): DotSimpleURL {
+        try {
+            const url = new URL(`http:${query}`);
+            return { host: url.host, pathname: url.pathname.substr(1) };
+        } catch {
+            return null;
+        }
     }
 
-    private validSearch(param: any): boolean {
-        return (
-            this.cleanInput(param).length &&
-            (!param.query.startsWith('//') ? param.query.length >= 3 : true)
-        );
+    private isSearchingForHost(query: string): boolean {
+        return query.startsWith('//') && !!this.getSiteName(query) ? !query.endsWith('/') : true;
     }
 
-    private cleanInput(event: any): string {
-        event.originalEvent.target.value = this.cleanQuery(event.query);
-        return event.originalEvent.target.value;
+    private getSiteName(site: string): string {
+        return site.replace(/\//g, '');
+    }
+
+    private resetResults(): void {
+        this.suggestions.next(of([]));
+    }
+
+    private cleanAndValidateQuery(query: string): string {
+        const cleanedQuery = this.cleanQuery(query);
+        this.autoComplete.inputEL.nativeElement.value = cleanedQuery;
```

Comment:
Don't we have this value on every key pressed in the events of this component?

Reply:
right, but I need that reference in other parts of the code so, is easier.